File: sdk/erlang/src/microsandbox.erl
-module(microsandbox).

-export([greet/1]).

%% @doc Returns a greeting message for the given name.
%%
%% Example:
%%   ```
%%   {ok, Greeting} = microsandbox:greet("World"),
%%   io:format("~s~n", [Greeting]).
%%   '''
-spec greet(string()) -> {ok, string()}.
greet(Name) ->
    Message = "Hello, " ++ Name ++ "! Welcome to Microsandbox!",
    io:format("~s~n", [Message]),
    {ok, Message}.

File: sdk/haskell/src/Microsandbox.hs
module Microsandbox
  ( greet
  ) where

-- | Returns a greeting message for the given name.
--
-- >>> greet "World"
-- "Hello, World! Welcome to Microsandbox!"
greet :: String -> IO String
greet name = do
  let message = "Hello, " ++ name ++ "! Welcome to Microsandbox!"
  putStrLn message
  return message
